src/dynamics/mpas/driver/dyn_mpas_subdriver.F90

```diff
@@ -703,37 +740,50 @@ end subroutine dyn_mpas_init_phase2
     !-------------------------------------------------------------------------------
     ! subroutine dyn_mpas_init_phase3
     !
-    !> \brief  Tracks `mpas_init` up to the point of calling `core_init`
+    !> \brief  Tracks `mpas_init` up to the point of calling `atm_core_init`
     !> \author Michael Duda
     !> \date   19 April 2019
     !> \details
     !>  This subroutine follows the stand-alone MPAS subdriver after the check on
     !>  the existence of the `streams.<core>` file up to, but not including,
-    !>  the point where `core_init` is called. It completes MPAS framework
+    !>  the point where `atm_core_init` is called. It completes MPAS framework
     !>  initialization, including the allocation of all blocks and fields managed
-    !>  by MPAS.
+    !>  by MPAS. However, scalars are allocated but not yet defined.
+    !>  `dyn_mpas_define_scalar` must be called afterwards. Also note that MPAS uses
+    !>  the term "scalar", but CAM-SIMA calls it "constituent".
     !> \addenda
     !>  Ported and refactored for CAM-SIMA. (KCW, 2024-03-06)
     !
     !-------------------------------------------------------------------------------
-    subroutine dyn_mpas_init_phase3(self, cam_pcnst, pio_file)
-        class(mpas_dynamical_core_type), intent(in) :: self
-        integer, intent(in) :: cam_pcnst
+    subroutine dyn_mpas_init_phase3(self, number_of_constituents, pio_file)
+        class(mpas_dynamical_core_type), intent(inout) :: self
+        integer, intent(in) :: number_of_constituents
         type(file_desc_t), pointer, intent(in) :: pio_file
 
         character(*), parameter :: subname = 'dyn_mpas_subdriver::dyn_mpas_init_phase3'
         character(strkind) :: mesh_filename
         integer :: mesh_format
+        integer, pointer :: num_scalars => null()
+        type(mpas_pool_type), pointer :: mpas_pool => null()
```

Comment:
Rather than initialize these here (adding the implicit `SAVE` attribute), is there a reason to not just nullify these either at the beginning of this routine or just before they are (hopefully) set? Would doing this remove the need to explicitly nullify these at the end of the subroutine?

Reply:
I pushed https://github.com/ESCOMP/CAM-SIMA/pull/267/commits/a731e9769498b5a3b374c2329aaf97eaec2ff853 to avoid explicit initialization of pointers.

Thanks for catching this. I don't know why I thought the rule doesn't apply to pointers.

Reply:
It's possible that I'm wrong, but `5.1 Type declaration statements" of the 2003 standard states:
```
33 If initialization is =>null-init, object-name shall be a pointer, and its initial association status is disas-
34 sociated.
35 The presence of initialization implies that object-name is saved, except for an object-name in a named
36 common block or an object-name with the PARAMETER attribute. The implied SAVE attribute may
37 be reaffirmed by explicit use of the SAVE attribute in the type declaration statement, by inclusion of
38 the object-name in a SAVE statement (5.2.12), or by the appearance of a SAVE statement without a
39 saved-entity -list in the same scoping unit.
```